media-sound/mixxx/mixxx-9999.ebuild

```
@@ -1,127 +1,180 @@
+	)
+	mp4? ( media-libs/libmp4v2 )
+	opus? (
+		media-libs/opus
+		media-libs/opusfile
+	)
+	qtkeychain? ( dev-libs/qtkeychain:=[qt6(-)] )
+	rubberband? ( media-libs/rubberband:= )
+	shout? ( dev-libs/openssl:= )
+	upower? (
+		dev-libs/glib:2
+		sys-power/upower:=
+	)
 	wavpack? ( media-sound/wavpack )
 "
 DEPEND="${RDEPEND}
 	dev-cpp/ms-gsl
+	test? ( dev-cpp/gtest )
 "
 BDEPEND="
-	dev-qt/qttest:5
-	dev-qt/qtxmlpatterns:5
+	dev-util/spirv-tools
 	virtual/pkgconfig
 "
+# Not todo to prevent links
+#benchmark? (
+#	dev-cpp/benchmark
+#	dev-util/google-perftools
+#)
 
 PATCHES=(
-	"${FILESDIR}"/${PN}-9999-docs.patch
+	# Building mixxx-test target only with explicit test useflag
+	"${FILESDIR}"/${P}-tests.patch
+	# Fix strict-aliasing violations in vendored katai_cpp_stl_runtime
+	# https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/commit/c01f530.patch
+	"${FILESDIR}"/${PN}-2.5.0-fix-strict-aliasing-kaitai.patch
 )
 
+CMAKE_SKIP_TESTS=(
+	# need HID controller
+	LegacyControllerMappingValidationTest.HidMappingsValid
+)
+
+PLOCALES="
+	ca cs de en es fi fr gl id it ja nl pl pt ro ru sl sq sr tr zh-CN zh-TW
+"
+
+mixxx_set_globals() {
+	local lang
+	local MANUAL_URI_BASE="https://downloads.mixxx.org/manual/${MY_PV}"
+	for lang in ${PLOCALES} ; do
+		SRC_URI+=" l10n_${lang}? ( ${MANUAL_URI_BASE}/${PN}-manual-${MY_PV}-${lang/-/_}.pdf )"
+		IUSE+=" l10n_${lang/ en/ +en}"
+	done
+	SRC_URI+=" ${MANUAL_URI_BASE}/${PN}-manual-${MY_PV}-en.pdf"
+}
+mixxx_set_globals
+
 src_configure() {
+	# prevent ld error as package builds static libs.
+	tc-is-lto && append-flags $(test-flags -ffat-lto-objects)
+
 	local mycmakeargs=(
-		-DFAAD="$(usex aac on off)"
-		-DFFMPEG="$(usex ffmpeg on off)"
-		-DHID="$(usex hid on off)"
-		-DLILV="$(usex lv2 on off)"
-		-DMAD="$(usex mp3 on off)"
-		-DOPTIMIZE="off"
-		-DCCACHE_SUPPORT="off"
-		-DOPUS="$(usex opus on off)"
-		-DBROADCAST="$(usex shout on off)"
-		-DVINYLCONTROL="on"
-		-DINSTALL_USER_UDEV_RULES=OFF
-		-DWAVPACK="$(usex wavpack on off)"
-		-DQTKEYCHAIN="$(usex qtkeychain on off)"
-		-DKEYFINDER="$(usex keyfinder on off)"
-		-DDOWNLOAD_MANUAL=OFF
+		-DBATTERY="$(usex upower ON OFF)"
+		-DBROADCAST="$(usex shout ON OFF)"
 		-DBUILD_SHARED_LIBS=OFF
+		-DBUILD_TESTING="$(usex test ON OFF)"
+		#-DBUILD_BENCH="$(usex benchmark ON OFF)"
+		# prevent duplicate call
+		-DCCACHE_SUPPORT=OFF
+		-DENGINEPRIME=OFF
+		-DFAAD="$(usex aac ON OFF)"
+		-DFFMPEG="$(usex ffmpeg ON OFF)"
+		#-DGPERFTOOLS="$(usex benchmark ON OFF)"
+		#-DGPERFTOOLSPROFILER="$(usex benchmark ON OFF)"
+		-DHID=ON
+		-DINSTALL_USER_UDEV_RULES=OFF
+		-DKEYFINDER="$(usex keyfinder ON OFF)"
+		-DLILV="$(usex lv2 ON OFF)"
+		-DMAD="$(usex mp3 ON OFF)"
+		-DMODPLUG="$(usex modplug ON OFF)"
+		-DOPTIMIZE=OFF
+		-DOPUS="$(usex opus ON OFF)"
+		-DPORTMIDI="$(usex midi ON OFF)"
+		-DQML="$(usex qml ON OFF)"
+		-DQTKEYCHAIN="$(usex qtkeychain ON OFF)"
+		-DRUBBERBAND="$(usex rubberband ON OFF)"
+		-DVINYLCONTROL=ON
+		-DWAVPACK="$(usex wavpack ON OFF)"
```

```suggestion
		-DBATTERY="$(usex upower)"
		-DBROADCAST="$(usex shout)"
		-DBUILD_SHARED_LIBS=OFF
		-DBUILD_TESTING="$(usex test)"
		#-DBUILD_BENCH="$(usex benchmark)"
		# prevent duplicate call
		-DCCACHE_SUPPORT=OFF
		-DENGINEPRIME=OFF
		-DFAAD="$(usex aac)"
		-DFFMPEG="$(usex ffmpeg)"
		#-DGPERFTOOLS="$(usex benchmark)"
		#-DGPERFTOOLSPROFILER="$(usex benchmark)"
		-DHID=ON
		-DINSTALL_USER_UDEV_RULES=OFF
		-DKEYFINDER="$(usex keyfinder)"
		-DLILV="$(usex lv2)"
		-DMAD="$(usex mp3)"
		-DMODPLUG="$(usex modplug)"
		-DOPTIMIZE=OFF
		-DOPUS="$(usex opus)"
		-DPORTMIDI="$(usex midi)"
		-DQML="$(usex qml)"
		-DQTKEYCHAIN="$(usex qtkeychain)"
		-DRUBBERBAND="$(usex rubberband)"
		-DVINYLCONTROL=ON
		-DWAVPACK="$(usex wavpack)"
```

The ON OFF dance is really not necessary, usex does it for us by default.